calyx/src/frontend/ast.rs

```diff
@@ -271,6 +271,8 @@ pub enum Control {
         attributes: ir::Attributes,
         /// Combinational group that may execute with this invoke.
         comb_group: Option<ir::Id>,
+        /// External cells that may execute with this invoke.
+        external_cells: Option<Vec<(ir::Id, ir::Id)>>,
```

Comment:
This doesn’t need to be an option. An empty vector can represent an absence of external cell inputs.